backend/tracim_backend/lib/core/application.py

```diff
@@ -6,33 +6,53 @@
 from tracim_backend.app_models.workspace_menu_entries import all_content_menu_entry
 from tracim_backend.app_models.workspace_menu_entries import dashboard_menu_entry
 from tracim_backend.exceptions import AppDoesNotExist
+from tracim_backend.lib.utils.app import TracimApplication
+from tracim_backend.lib.utils.app import TracimContentType
 
 if typing.TYPE_CHECKING:
     from tracim_backend.models.data import Workspace
-    from tracim_backend.app_models.applications import Application  # noqa:F401
-    from tracim_backend.app_models.contents import ContentType  # noqa: F401
+    from tracim_backend.config import CFG
+    from tracim_backend.app_models.applications import TracimApplicationInContext  # noqa:F401
+    from tracim_backend.app_models.contents import ContentTypeInContext  # noqa: F401
 
 
 class ApplicationApi(object):
-    def __init__(self, app_list: List["Application"], show_all: bool = False) -> None:
+    def __init__(self, app_list: List["TracimApplication"], show_all: bool = False) -> None:
         self.apps = app_list
         self.show_all = show_all
 
-    def get_one(self, slug):
+    def exist(self, slug: str) -> bool:
+        """ Check if app with this slug does exist according to applicationApi configuration"""
+        try:
+            self.get_one(slug)
+            return True
+        except AppDoesNotExist:
+            return False
+
+    def get_one(self, slug: str) -> TracimApplication:
+        """ Get app with given slug if exist"""
         for app in self.apps:
             if app.slug == slug:
-                return app
+                if self.show_all or app.is_active:
+                    return app
         raise AppDoesNotExist("Application {app} does not exist".format(app=slug))
 
-    def get_all(self) -> List["Application"]:
+    def get_application_in_context(
+        self, app: TracimApplication, app_config: "CFG"
+    ) -> "TracimApplicationInContext":
+        from tracim_backend.app_models.applications import TracimApplicationInContext
```

Comment:
Wrong place import (or there is a reason ?)

Reply:
If cyclic import or goo reason: add a note

Reply:
yes, circular import, i added a comment.